### PR TITLE
Enable lint checks on test sources

### PR DESCRIPTION
**Pull request by shaishavgandhi05**
This also enables IDE checks for test sources.

screen shot 2018-12-02 at 10 15 25 pm

### CODE REVIEW

File: static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt [hunk 1]
```diff
@@ -46,7 +46,11 @@ class AutoDisposeDetector: Detector(), SourceCodeScanner {
         Category.CORRECTNESS,
         10,
         Severity.ERROR,
-        Implementation(AutoDisposeDetector::class.java, EnumSet.of(Scope.JAVA_FILE)))
+        Implementation(AutoDisposeDetector::class.java,
```
Comment (hzsweers) [inline static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt]: Before merging, can you add a comment explaining why we have to do it this way?

File: static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt [hunk 2]
```diff
@@ -46,6 +46,11 @@ class AutoDisposeDetector: Detector(), SourceCodeScanner {
         Category.CORRECTNESS,
         10,
         Severity.ERROR,
+        // We use the overloaded constructor that takes a varargs of `Scope` as the last param.
+        // This is to enable on-the-fly IDE checks. We are telling lint to run on both
+        // JAVA and TEST_SOURCES in the `scope` parameter but by providing the `analysisScopes`
+        // params, we're indicating that this check can run on either JAVA or TEST_SOURCES and
+        // doesn't require both of them together.
```
Comment (hzsweers) [inline static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt]: Link the source on this in lint-dev? Also can we remove when we have min AGP 3.3?

Comment (shaishavgandhi05) [reply static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt]: You mean 3.4? Tor mentioned that his changes [will land in 3.4](https://groups.google.com/d/msg/lint-dev/ULQMzW1ZlP0/1dG4Vj3-AQAJ)

Comment (hzsweers) [reply static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt]: errr yes 3.4 :)

Comment (shaishavgandhi05) [reply static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt]: 9276702618beb53936da4cec376d238a8d54ded6

File: static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt (hunk 1 above)
Comment (shaishavgandhi05) [reply static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt]: b7df26136f0822c527f1de85cb0b251a26c8fad3

**Pull request merged**